shared/compression_test.go

```diff
@@ -0,0 +1,229 @@
+package shared
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/klauspost/compress/gzip"
+)
+
+func TestCompressionTransport_Gzip(t *testing.T) {
+	// Create a test server that verifies the request is compressed
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Content-Encoding") != "gzip" {
+			t.Error("Expected Content-Encoding: gzip header")
+		}
+
+		// Decompress and verify the body
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil {
+			t.Fatalf("Failed to create gzip reader: %v", err)
+		}
+		defer func() { _ = gzipReader.Close() }()
+
+		body, err := io.ReadAll(gzipReader)
+		if err != nil {
+			t.Fatalf("Failed to read decompressed body: %v", err)
+		}
+
+		expectedBody := "test request body"
+		if string(body) != expectedBody {
+			t.Errorf("Expected body %q, got %q", expectedBody, string(body))
+		}
+
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write([]byte("OK"))
+		if err != nil {
+			t.Fatalf("Failed to write response: %v", err)
+		}
+	}))
+	defer server.Close()
+
+	// Create compression transport
+	transport := NewCompressionTransport(http.DefaultTransport, NewGzipConfig().GzipRequestCompressor())
+
+	// Create a request with a body
+	req, err := http.NewRequest("POST", server.URL, strings.NewReader("test request body"))
+	if err != nil {
+		t.Fatalf("Failed to create request: %v", err)
+	}
+
+	// Execute the request
+	resp, err := transport.RoundTrip(req)
+	if err != nil {
+		t.Fatalf("RoundTrip failed: %v", err)
+	}
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Expected status 200, got %d", resp.StatusCode)
+	}
+}
+
+func TestCompressionTransport_NoBody(t *testing.T) {
```

Comment:
Could you please do parametric test with no body, small body, large body.
Pseudo-random-genrate body based with const seed, test content lengh, size of the body, unpack it and chec if it matches what was sent over.